subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/AbstractRecompilationSpecProvider.java

```diff
@@ -105,20 +105,6 @@ protected void includePreviousCompilationOutputOnClasspath(JavaCompileSpec spec)
     }
 
     protected String rebuildClauseForChangedNonSourceFile(String description, FileChange fileChange) {
-        String changeName = determineChangeName(fileChange.getChangeType());
-        return description + " '" + fileChange.getFile().getName() + "' has been " + changeName;
-    }
-
-    private String determineChangeName(ChangeType changeType) {
-        switch (changeType) {
-            case ADDED:
-                return "added";
-            case REMOVED:
-                return "removed";
-            case MODIFIED:
-                return "modified";
-            default:
-                throw new AssertionError("Unknown change type: " + changeType);
-        }
+        return description + " '" + fileChange.getFile().getName() + "' has been " + fileChange.getChangeType().name().toLowerCase(Locale.US);
```

Comment:
Nit: let's turn this into `String.format()` or something.

Reply:
Done.